app/models/services/index.ts

```diff
@@ -142,7 +142,13 @@ function isCollectionMatch(
   operation: DataOperation,
   serviceConfig: ServiceConfig<unknown>,
 ): boolean {
-  return operation.sources.every((source) => serviceConfig.collections.includes(source.collection));
+  return operation.sources.every((source) => {
+    let matchedCollection;
+    if ( serviceConfig.collections.includes(source.collection) ) return true;
+    else if ( matchedCollection = serviceConfig.collections.find(x => x[source.collection]) ) {
+      return source.variables.every(x => matchedCollection[source.collection].includes(x.id));
```

Comment:
What about requests that do not request variable subsetting (ones that specify 'all')? Can we pass all to Giovanni? Will Giovanni just return the variables it allows?

Reply:
Yes, Giovanni is variable-based service, and we have to specify the variables for the request.

Reply:
After a walkthrough, we found that when `all` is specified from the request, `source.variables` will be `undef` and an exception will be raised.
This is fixed from more recent commits (along with test added).